content/en/agent/logs/proxy.md

```diff
@@ -60,17 +69,219 @@ The parameter above can also be set with the following environment variable:
+The file might be located at `/etc/ssl/certs/ca-bundle.crt` for CentOS, Redhat.
+
+Once the HAProxy configuration is in place, you can reload it or restart HAProxy. **It is recommended to have a `cron` job that reloads HAProxy every 10 minutes** (usually doing something like `service haproxy reload`) to force a refresh of HAProxy's DNS cache, in case `app.datadoghq.com` fails over to another IP.
+
+{{% /tab %}}
+{{% tab "Datadog EU site" %}}
+
+```conf
+# Basic configuration
+global
+    log 127.0.0.1 local0
+    maxconn 4096
+    stats socket /tmp/haproxy
 
-When the Agent is [configured to send logs through HTTPS][1], use the same [set of proxy settings][2] as the other data types in order to send logs through a web proxy.
+# Some sane defaults
+defaults
+    log     global
+    option  dontlognull
+    retries 3
+    option  redispatch
+    timeout client 5s
+    timeout server 5s
+    timeout connect 5s
+
+# This declares a view into HAProxy statistics, on port 3833
+# You do not need credentials to view this page and you can
+# turn it off once you are done with setup.
+listen stats
+    bind *:3833
+    mode http
+    stats enable
+    stats uri /
+    
+# This declares the endpoint where your Agents connects for
+# sending Logs (e.g the value of "logs.config.logs_dd_url")
+frontend logs_frontend
+    bind *:10514
+    mode tcp
+    default_backend datadog-logs
+    
+# This is the Datadog server. In effect any TCP request coming
+# to the forwarder frontends defined above are proxied to
+# Datadog's public endpoints.
+
+backend datadog-logs
+    balance roundrobin
+    mode tcp
+    option tcplog
+    server datadog agent-intake.logs.datadoghq.eu:443 ssl verify required ca-file /etc/ssl/certs/ca-bundle.crt check port 443
+```
+
+**Note**: Download the certificate with the following command:
+
+* `sudo apt-get install ca-certificates` (Debian, Ubuntu)
+* `yum install ca-certificates` (CentOS, Redhat)
+
+The file might be located at `/etc/ssl/certs/ca-bundle.crt` for CentOS, Redhat.
+
+Once the HAProxy configuration is in place, you can reload it or restart HAProxy. **It is recommended to have a `cron` job that reloads HAProxy every 10 minutes** (usually doing something like `service haproxy reload`) to force a refresh of HAProxy's DNS cache, in case `app.datadoghq.eu` fails over to another IP.
 
-[1]: /agent/logs/?tab=tailexistingfiles#send-logs-over-https
-[2]: /agent/proxy
 {{% /tab %}}
 {{< /tabs >}}
 
+#### Agent configuration
+
+Then edit the `datadog.yaml` Agent configuration file and set `logs_no_ssl` to `true`. This is needed as Haproxy does not simply forward the traffic and is not the Datadog backend so cannot use the same certificate.
+
+**Note**: `logs_no_ssl` might set to true because Haproxy is configured to encrypt the data. Do not set this parameter to `true` otherwise.
+
+```
+logs_config:
+  logs_dd_url: "<PROXY_SERVER_DOMAIN>:10514"
+  logs_no_ssl: true
+```
+
+### Using NGINX as a TCP Proxy for logs
```

Comment:
How about adding the HTTP counter-part  section? 
The previous nginx configuration based on **_streams_** works well for both TCP and HTTP and is very similar to what is done with HAProxy. 
But in many nginx implementations we see people using **_http_** blocks in their configs hence this suggestion:
This would assume using the following type of settings:


**1. datadog.yaml**

```    
dd_url: https://6-15-1-app.agent.datadoghq.com
proxy:
  https: http://10.0.0.249:8888
  http: http://10.0.0.249:8888

skip_ssl_validation: true
logs_enabled: true
logs_config:
  logs_dd_url: agent-http-intake.logs.datadoghq.com:443 # HTTP direct
  use_http: true
  use_compression: true
  compression_level: 6
```

**2. nginx.conf**

```
user www-data;
worker_processes auto;
events { }

http {
    server_names_hash_bucket_size 128;

    access_log /var/log/nginx_access.log;
    error_log /var/log/nginx_errors.log;

    # Whitelist dataodghq.com and datadoghq.eu
    server {
        listen       8888;
        server_name *.datadoghq.com;
        server_name *.datadoghq.eu;
        #server_name ~^(.+)$; #catch all 
        proxy_connect;
        proxy_max_temp_file_size 0;
        resolver 8.8.8.8;
        location / {
           proxy_pass http://$http_host;
           proxy_set_header Host $http_host;
        }
    }

    # Disable all other domains
    server {
        listen       8888;
        server_name ~.+;
        return 404;
    }

}
```
And last but not least, using the **_ngx_http_proxy_connect_module_** as per 
https://github.com/chobits/ngx_http_proxy_connect_module

**3. ngx_http_proxy_connect_module**

Nginx is can be configured for forwarding proxying. Unfortunately, that doesn't work very well with HTTPS connections. As soon the user is calling a URL via https, Nginx will throw errors. There is a StackOverflow issue to that topic (https://superuser.com/questions/604352/nginx-as-forward-proxy-for-https). In short SSL Forwarding doesn't  work well when trying to do it using http block
(That said It does when using stream block).
 
Luckily there is a solution for that problem. The ngx_http_proxy_connect_module is solving this issue. If Nginx is compiled with that module, the proxying will work with SSL connections as well.


```
ubuntu@ip-10-0-0-249:~$ wget https://nginx.org/download/nginx-1.14.2.tar.gz
ubuntu@ip-10-0-0-249:~$ tar xvf nginx-1.14.2.tar.gz
ubuntu@ip-10-0-0-249:~$ cd nginx-1.14.2
ubuntu@ip-10-0-0-249:~/nginx-1.14.2 $ git clone https://github.com/chobits/ngx_http_proxy_connect_module
ubuntu@ip-10-0-0-249:~/nginx-1.14.2 $ patch -p1 < ~/ngx_http_proxy_connect_module/patch/proxy_connect_1014.patch
ubuntu@ip-10-0-0-249:~/nginx-1.14.2 $ ./configure --add-module=./ngx_http_proxy_connect_module
ubuntu@ip-10-0-0-249:~/nginx-1.14.2 $ make && make install
```
From there the newly built nginx version can be used along with the configuration files (datadog.yaml and nginx.conf)

**4. Additional resources**
Boyan's great contributions:
- Configs (Agent + nginx) using stream  : https://github.com/ncracker/dd-nginx-proxy
- Docker image : https://hub.docker.com/repository/docker/ncracker/dd-nginx-proxy

Misiu  & Oskar 
- Docker image with the above patched nginx: https://github.com/DataDog/Miscellany

Reply:
Thanks for this.

Let's sync internally on this.